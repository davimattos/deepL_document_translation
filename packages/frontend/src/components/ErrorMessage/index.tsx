import { AlertCircle } from "lucide-react";
import { Props } from "./types";

export function ErrorMessage({ errorMessage }: Props) {
  return (
    <div className="bg-red-50 border border-red-200 rounded-lg p-4">
      <div className="flex items-start gap-3">
        <AlertCircle
          data-testid="alert-icon"
          className="w-5 h-5 text-red-500 mt-0.5 flex-shrink-0"
        />
        <p className="text-red-700 text-sm">{errorMessage}</p>
      </div>
    </div>
  );
}
