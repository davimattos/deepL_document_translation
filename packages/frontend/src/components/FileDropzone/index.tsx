import { Upload, CheckCircle } from "lucide-react";
import { Props } from "./types";

export function FileDropzone({
  handleDrop,
  handleDragOver,
  file,
  fileInputRef,
  handleFileSelect,
  allowedExtensions,
}: Props) {
  return (
    <div className="p-8">
      <div
        onDrop={handleDrop}
        onDragOver={handleDragOver}
        className={`border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 ${
          file
            ? "border-green-300 bg-green-50"
            : "border-gray-300 bg-gray-50 hover:border-blue-400 hover:bg-blue-50"
        }`}
      >
        <input
          ref={fileInputRef}
          type="file"
          accept={allowedExtensions.toString()}
          onChange={e => e.target.files?.[0] && handleFileSelect(e.target.files[0])}
          className="hidden"
        />

        {file ? (
          <div className="space-y-4">
            <CheckCircle className="w-16 h-16 text-green-500 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">Arquivo Selecionado</h3>
              <p className="text-gray-600 mb-1">{file.name}</p>
              <p className="text-sm text-gray-500">{(file.size / 1024 / 1024).toFixed(2)} MB</p>
            </div>
            <button
              onClick={() => fileInputRef.current?.click()}
              className="text-blue-600 hover:text-blue-700 font-medium transition-colors"
            >
              Selecionar outro arquivo
            </button>
          </div>
        ) : (
          <div className="space-y-4">
            <Upload className="w-16 h-16 text-gray-400 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">Carregar Documento</h3>
              <p className="text-gray-600 mb-4">
                Arraste e solte seu arquivo aqui ou clique para selecionar
              </p>
              <button
                onClick={() => fileInputRef.current?.click()}
                className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-3 rounded-lg font-medium transition-colors"
              >
                Selecionar Arquivo
              </button>
            </div>
            <div className="text-sm text-gray-500">
              <p className="mb-2">Extens√µes suportadas:</p>
              <p className="font-mono text-xs">{allowedExtensions.join(", ")}</p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
