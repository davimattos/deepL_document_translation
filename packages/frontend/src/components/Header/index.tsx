import { FileText } from "lucide-react";

export function Header() {
  return (
    <div className="text-center mb-12">
      <div className="flex items-center justify-center mb-6">
        <div className="bg-blue-600 p-3 rounded-xl">
          <FileText role="img" aria-label="Icone de documento" className="w-8 h-8 text-white" />
        </div>
      </div>
      <h1 className="text-4xl font-bold text-gray-900 mb-4">
        Tradutor de Documentos
      </h1>
      <p className="text-xl text-gray-600 max-w-2xl mx-auto">
        Traduza seus documentos usando a API do DeepL
      </p>
    </div>
  );
}
