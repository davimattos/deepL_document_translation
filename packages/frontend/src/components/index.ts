export * from './Header'
export * from './LanguageSelector'
export * from './FileDropzone'
export * from './ProcessingProgress'
export * from './InfoCard'
export * from './ErrorMessage'