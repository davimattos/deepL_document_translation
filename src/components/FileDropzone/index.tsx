import { Upload, CheckCircle } from "lucide-react";
import { Props } from "./types";

export function FileDropzone({
  handleDrop,
  handleDragOver,
  file,
  fileInputRef,
  handleFileSelect,
  allowedExtensions,
  handleExecutedFile
}: Props) {

  return (
    <div className="p-8">
      <div
        onDrop={handleDrop}
        onDragOver={handleDragOver}
        className={`border-2 border-dashed rounded-xl p-12 text-center transition-all duration-300 ${
          file
            ? "border-green-300 bg-green-50"
            : "border-gray-300 bg-gray-50 hover:border-blue-400 hover:bg-blue-50"
        }`}
      >
        <input
          ref={fileInputRef}
          type="file"
          accept=".docx,.pptx,.xlsx,.pdf,.htm,.html,.txt,.xlf,.xliff,.srt"
          onChange={(e) =>
            e.target.files?.[0] && handleFileSelect(e.target.files[0])
          }
          className="hidden"
        />

        {file ? (
          <div className="space-y-4">
            <CheckCircle className="w-16 h-16 text-green-500 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">
                Arquivo Selecionado
              </h3>
              <p className="text-gray-600 mb-1">{file.name}</p>
              <p className="text-sm text-gray-500">
                {(file.size / 1024 / 1024).toFixed(2)} MB
              </p>
            </div>
            <button
              onClick={handleExecutedFile}
              className="text-blue-600 hover:text-blue-700 font-medium transition-colors"
            >
              Selecionar outro arquivo
            </button>
          </div>
        ) : (
          <div className="space-y-4">
            <Upload className="w-16 h-16 text-gray-400 mx-auto" />
            <div>
              <h3 className="text-xl font-semibold text-gray-900 mb-2">
                Carregar Documento
              </h3>
              <p className="text-gray-600 mb-4">
                Arraste e solte seu arquivo aqui ou clique para selecionar
              </p>
              <button
                onClick={() => fileInputRef.current?.click()}
                className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-3 rounded-lg font-medium transition-colors"
              >
                Selecionar Arquivo
              </button>
            </div>
            <div className="text-sm text-gray-500">
              <p className="mb-2">Extens√µes suportadas:</p>
              <p className="font-mono text-xs">
                {allowedExtensions.join(", ")}
              </p>
            </div>

            {/* File Format Limits Table */}
            <div className="mt-6 bg-gray-50 rounded-lg p-4">
              <h4 className="font-semibold text-gray-900 mb-3">
                Limites por Formato de Arquivo
              </h4>
              <div className="overflow-x-auto">
                <table className="w-full text-xs">
                  <thead>
                    <tr className="border-b border-gray-200">
                      <th className="text-center py-2 font-medium text-gray-700">
                        Formato
                      </th>
                      <th className="text-center py-2 font-medium text-gray-700">
                        DeepL Suporte
                      </th>
                    </tr>
                  </thead>
                  <tbody className="text-gray-600">
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Word (.docx / .doc) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">PowerPoint (.pptx) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Excel (.xlsx) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">PDF (.pdf) </td>
                      <td className="text-center py-2">
                        30 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">Text (.txt)</td>
                      <td className="text-center py-2">
                        1 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">HTML (.html)</td>
                      <td className="text-center py-2">
                        5 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr className="border-b border-gray-100">
                      <td className="py-2 font-mono">XLIFF (.xlf/.xliff)*</td>
                      <td className="text-center py-2">
                        10 MB
                        <br />
                        1M chars
                      </td>
                    </tr>
                    <tr>
                      <td className="py-2 font-mono">SRT (.srt) </td>
                      <td className="text-center py-2">
                        150 KB
                        <br />
                        1M chars
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <p className="text-xs text-gray-500 mt-3">
                * Os limites aplicam-se por documento individual
              </p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
