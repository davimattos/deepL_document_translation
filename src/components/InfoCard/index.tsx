import { Props } from "./types";

export function InfoCard({icon, title, description, className = 'bg-gray-100'}: Props) {
  return (
    <div className="bg-white p-6 rounded-xl shadow-sm border border-gray-100">
      <div className={`w-12 h-12 rounded-lg flex items-center justify-center mb-4 ${className}`}>
        {icon}
      </div>
      <h3 className="font-semibold text-gray-900 mb-2">{title}</h3>
      <p className="text-gray-600 text-sm">
        {description}
      </p>
    </div>
  );
}
