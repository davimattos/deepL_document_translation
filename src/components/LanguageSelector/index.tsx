import { Props } from "./types";

export function LanguageSelector({
  languages,
  sourceLanguages,
  source,
  target
}: Props) {
  return (
    <div className="bg-white rounded-2xl shadow-xl border border-gray-100 overflow-hidden mb-8">
      <div className="p-8">
        <h2 className="text-2xl font-semibold text-gray-900 mb-6">
          Configuração
        </h2>

        <div className="grid md:grid-cols-2 gap-6">
          <div>
            <label
              htmlFor="sourceLanguage"
              className="block text-sm font-medium text-gray-700 mb-2"
            >
              Idioma de Origem
            </label>
            <select
              id="sourceLanguage"
              value={source.sourceLanguage}
              onChange={(e) => source.setSourceLanguage(e.target.value)}
              className="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-colors"
            >
              {sourceLanguages.map((lang: any) => (
                <option key={lang.code} value={lang.code}>
                  {lang.name}
                </option>
              ))}
            </select>
            <p className="text-xs text-gray-500 mt-1">
              Selecione o idioma do documento original
            </p>
          </div>

          <div>
            <label
              htmlFor="targetLanguage"
              className="block text-sm font-medium text-gray-700 mb-2"
            >
              Idioma de Destino
            </label>
            <select
              id="targetLanguage"
              value={target.targetLanguage}
              onChange={(e) => target.setTargetLanguage(e.target.value)}
              className="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-colors"
            >
              {languages.map((lang: any) => (
                <option key={lang.code} value={lang.code}>
                  {lang.name}
                </option>
              ))}
            </select>
          </div>
        </div>
      </div>
    </div>
  );
}
