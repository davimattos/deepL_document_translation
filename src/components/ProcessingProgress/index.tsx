import { FileText, Loader2, AlertCircle } from "lucide-react";
import { ErrorMessage } from "../ErrorMessage";
import { Props } from "./types";

export function ProcessingProgress({
  processing,
  processFile,
  apiKey,
  setProcessing,
  setErrorMessage,
  errorMessage,
}: Props) {
  return (
    <div className="border-t border-gray-100 p-8">
      <div className="space-y-6">
        {/* Progress Bar */}
        {processing.status !== "idle" && (
          <div className="space-y-3">
            <div className="flex items-center justify-between">
              <span className="text-sm font-medium text-gray-700">
                {processing.message}
              </span>
              <span className="text-sm text-gray-500">
                {processing.progress}%
              </span>
            </div>
            <div
              className="w-full bg-gray-200 rounded-full h-2"
              role="progressbar"
              aria-valuemin={0}
              aria-valuemax={100}
              aria-valuenow={processing.progress}
              aria-label="Progresso de tradução"
            >
              <div
                className={`h-2 rounded-full transition-all duration-500 ${
                  processing.status === "error"
                    ? "bg-red-500"
                    : processing.status === "completed"
                    ? "bg-green-500"
                    : "bg-blue-500"
                }`}
                style={{ width: `${processing.progress}%` }}
              />
            </div>
          </div>
        )}

        {/* Action Buttons */}
        <div className="flex items-center gap-4">
          {processing.status === "idle" && (
            <button
              onClick={processFile}
              disabled={!apiKey.trim()}
              className={`flex-1 px-6 py-3 rounded-lg font-medium transition-colors flex items-center justify-center gap-2 ${
                apiKey.trim()
                  ? "bg-blue-600 hover:bg-blue-700 text-white"
                  : "bg-gray-300 text-gray-500 cursor-not-allowed"
              }`}
            >
              <FileText className="w-5 h-5" />
              Traduzir com DeepL
            </button>
          )}

          {(processing.status === "uploading" ||
            processing.status === "processing") && (
            <button
              disabled
              className="flex-1 bg-blue-400 text-white px-6 py-3 rounded-lg font-medium flex items-center justify-center gap-2"
            >
              <Loader2 className="w-5 h-5 animate-spin" />
              Processando...
            </button>
          )}

          {processing.status === "error" && (
            <button
              onClick={() => {
                setProcessing({
                  status: "idle",
                  progress: 0,
                  message: "",
                });
                setErrorMessage("");
              }}
              className="flex-1 bg-red-600 hover:bg-red-700 text-white px-6 py-3 rounded-lg font-medium transition-colors flex items-center justify-center gap-2"
            >
              <AlertCircle className="w-5 h-5" />
              Tentar Novamente
            </button>
          )}
        </div>

        {/* Error Message */}
        {errorMessage && <ErrorMessage errorMessage={errorMessage} />}
      </div>
    </div>
  );
}
