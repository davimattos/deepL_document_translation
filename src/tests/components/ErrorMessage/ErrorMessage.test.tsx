import { render, screen } from "@testing-library/react";

import { ErrorMessage } from "../../../components/ErrorMessage";

describe("ErrorMessage component", () => {
  const errorText = "Something went wrong";

  it("renders the error message", () => {
    render(<ErrorMessage errorMessage={errorText} />);

    const paragraph = screen.getByText(errorText);
    expect(paragraph).toBeInTheDocument();
    expect(paragraph).toHaveTextContent(errorText);
    expect(paragraph).toHaveClass("text-red-700");
  });

  it("renders the alert icon with correct role", () => {
    render(<ErrorMessage errorMessage={errorText} />);
    
    const svgIcon = screen.getByTestId("alert-icon");
    expect(svgIcon).toBeInTheDocument();
    expect(svgIcon).toHaveClass("text-red-500");
  });

  it("matches snapshot", () => {
    const { container } = render(<ErrorMessage errorMessage={errorText} />);
    expect(container).toMatchSnapshot();
  });
});
