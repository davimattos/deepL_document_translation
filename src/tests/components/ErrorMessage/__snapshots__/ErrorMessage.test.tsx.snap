// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage component > matches snapshot 1`] = `
<div>
  <div
    class="bg-red-50 border border-red-200 rounded-lg p-4"
  >
    <div
      class="flex items-start gap-3"
    >
      <svg
        class="lucide lucide-alert-circle w-5 h-5 text-red-500 mt-0.5 flex-shrink-0"
        data-testid="alert-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="8"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="16"
          y2="16"
        />
      </svg>
      <p
        class="text-red-700 text-sm"
      >
        Something went wrong
      </p>
    </div>
  </div>
</div>
`;
